### PR TITLE
chore(slack): remove old slack client code for issue alerts

### PR DESCRIPTION
**Pull request by cathteng**
Default to using the Slack SDK Client for sending issue alert notifications in a channel. The feature has been GA'd.

### CODE REVIEW

**Review by iamrajjoshi — APPROVED** [review]

File: src/sentry/integrations/slack/metrics.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# metrics constants
+
+SLACK_ISSUE_ALERT_SUCCESS_DATADOG_METRIC = "sentry.integrations.slack.issue_alert.success"
+SLACK_ISSUE_ALERT_FAILURE_DATADOG_METRIC = "sentry.integrations.slack.issue_alert.failure"
```
Comment (cathteng) [inline src/sentry/integrations/slack/metrics.py:4]: adding metrics, i thought to dump them here

File: src/sentry/integrations/slack/actions/notification.py [hunk 2]
```diff
@@ -162,106 +154,70 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
...
+                    thread_ts=thread_ts,
+                    reply_broadcast=reply_broadcast,
+                )
+                metrics.incr(
+                    SLACK_ISSUE_ALERT_SUCCESS_DATADOG_METRIC,
+                    sample_rate=1.0,
+                    tags={
+                        "action": "send_notification",
+                        "ok": response.get("ok", False),
+                        "status": response.status_code,
```
Comment (cathteng) [inline src/sentry/integrations/slack/actions/notification.py:179]: maybe i don't need to be logging these for success?

File: src/sentry/integrations/slack/actions/notification.py [hunk 3]
```diff
@@ -162,106 +154,70 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
...
+                    unfurl_media=False,
+                    thread_ts=thread_ts,
+                    reply_broadcast=reply_broadcast,
+                )
+                metrics.incr(
+                    SLACK_ISSUE_ALERT_SUCCESS_DATADOG_METRIC,
+                    sample_rate=1.0,
+                    tags={
+                        "action": "send_notification",
+                        "ok": response.get("ok", False),
```
Comment (ykamo001) [inline src/sentry/integrations/slack/actions/notification.py:178]: do we need this tag if the metric name is success?

File: src/sentry/integrations/slack/actions/notification.py (hunk 2 above)
Comment (ykamo001) [reply src/sentry/integrations/slack/actions/notification.py:179]: does the status code matter for us on success?

File: src/sentry/integrations/slack/metrics.py (hunk 1 above)
Comment (ykamo001) [reply src/sentry/integrations/slack/metrics.py:4]: Love this pattern, thank you for adding and using constant!

**Review by ykamo001 — APPROVED** [review]
Approving to unblock, we can continue to discuss metrics and tags

File: src/sentry/integrations/slack/actions/notification.py (hunk 2 above)
Comment (cathteng) [reply src/sentry/integrations/slack/actions/notification.py:179]: no :P

File: src/sentry/integrations/slack/actions/notification.py (hunk 3 above)
Comment (cathteng) [reply src/sentry/integrations/slack/actions/notification.py:178]: no lol

**Pull request merged**